kernel/src/process/posix_thread/mod.rs

```diff
@@ -242,20 +224,22 @@ impl PosixThread {
 
     /// Enqueues a thread-directed signal.
     ///
-    /// This method does not perform permission checks on user signals. Therefore, unless the
-    /// caller can ensure that there are no permission issues, this method should be used for
-    /// enqueue kernel signals or fault signals.
+    /// This method does not perform permission checks on user signals.
+    /// Therefore, unless the caller can ensure that there are no permission issues,
+    /// this method should be used to enqueue kernel signals or fault signals.
     pub fn enqueue_signal(&self, signal: Box<dyn Signal>) {
         let process = self.process();
         let sig_dispositions = process.sig_dispositions().lock();
         let sig_dispositions = sig_dispositions.lock();
 
-        let signum = signal.num();
-        if sig_dispositions.get(signum).will_ignore(signum) {
+        let will_ignore = sig_dispositions.will_ignore(signal.as_ref());
+        let blocked = self.has_signal_blocked(signal.num());
```

Comment:
But this suffers from the same race condition as I described when enqueuing the process-wide signals?